docs/concepts/registries.md

```diff
@@ -40,63 +28,136 @@ public ExampleMod() {
 }
 ```
 
-Creating Registries
--------------------
+### `Register` events
+
+The `RegistryEvent`s are the second and more flexible way to register objects. These [events][] are fired after the mod constructors and before the loading of configs.
+
+The event used in registering objects is the `RegistryEvent.Register<T>`. The type parameter `T` should be set to the type of the object being registered. Calling `#getRegistry` will return the registry, upon which objects are registered with `#register` or `#registerAll`. 
+
+`RegistryEvent.Register<?>` events are fired in this order: first, the `Block` registry, then the `Item` registry, and then all other registries in alphabetical order. 
```

Comment:
> and then all other registries in alphabetical order. 

While Block and Item are first the other registries are not ordered in any way. Mods shouldn't break even if the registry events were called in a random order every startup.

Reply:
```java
// GameData.class, fireRegistryEvents(), L951-952, MC 1.14.4, mappings 20190719-1.14.3
// GameData.class, fireRegistryEvents(), L978-979, MC 1.15.2, mappings 20200514-1.15.1
List<ResourceLocation> keys = Lists.newArrayList(RegistryManager.ACTIVE.registries.keySet());
keys.sort((o1, o2) -> String.valueOf(o1).compareToIgnoreCase(String.valueOf(o2)));
```
The javadoc for `String#compareToIgnoreCase` is:
 > Compares two strings lexicographically, ignoring case differences.

So, current registry behavior is that Block and Item are first (evidenced by the lines following the aforementioned code), then all other registries in lexicographical/alphabetical order. This is supported by looking at run logs, which give the stated order (`minecraft:block`, `minecraft:item`, the `forge` registries in alphabetical order, then the `minecraft` registries in alphabetical order). Forge registries run before Minecraft ones because `f`orge is higher than `m`inecraft.

I somewhat agree that mods shoudn't break when registry events after Block and Item are random. However, by having the registries load in lexicographical order, it gives modders confidence that this registry will *always* be run before another. I don't know how that knowledge will benefit the modder, but at least they have it.

Reply:
That is the problem: you shouldn't give modders the confidence that registries will load in a certain order, since registries should be completely indipendent of each other. Moreover, as outlined in FAP20, there was an idea to make registry events parallel, meaning that multiple registry events may fire at the same time on different threads.

Reply:
But that idea of parallel registry events are still that: an idea. Until Forge changes from the current behavior, we should still document the current behavior.

Once parallel registry events are introduced and registry loader order cannot be guaranteed, this line will be removed. For now, it exists to inform the modder that there is an load order: Blocks, then Items, then everything else.

To me, it also guides a person from deducing: "So, Blocks are always after Items? So, I can at least reference Blocks in the Items registry event load."

Reply:
My comment was based on FAP20 where they talked about how the registries could be called in parallel. Them running in alphabetical order is not a feature, it's just how it happens to work. If it's noted in the documentation someone might base their code on it and find out in a few versions it doesn't work anymore. I think we either shouldn't mention the order in which they are called at all or already mention one should assume the order is random.

Reply:
So, after re-watching that part of FAP20, it does seem that random registry load order may happen. Since I do not want to put an assumption into documentation that currently contradicts current behavior, I'll remove that paragraph altogether.